sale_product_deposit/models/product_template.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    total_with_vat = fields.Float(
+        compute="_compute_total",
+        store=True,
+        string="Total Sales Price with VAT",
+    )
+    total_with_vat_by_unit = fields.Float(
+        compute="_compute_total",
+        store=True,
+        string="Total Sales Price with VAT by Reference Unit",
+    )
+    total_deposit = fields.Float(
+        compute="_compute_total", store=True, string="Deposit Price"
+    )
+    several_tax_strategies_warning = fields.Boolean(
+        string="This product can't be printed from the Point"
+        " of Sale because several tax strategies were defined.",
+        compute="_compute_total",
+    )
+
+    @api.depends(
+        "taxes_id",
+        "list_price",
+        "taxes_id.amount",
+        "taxes_id.tax_group_id",
+        "weight",
+    )
+    def _compute_total(self):
+        for product in self:
+            product.total_with_vat = False
+            product.total_with_vat_by_unit = False
+            product.total_deposit = False
+            deposit_group = self.env.ref(
+                "sale_product_deposit.deposit_tax_group", raise_if_not_found=False
+            )
+            product.several_tax_strategies_warning = False
```

Comment:
i don’t like the fact to change these values several times. this can possibly have bad effects (computed fields depending on them being recomputed multiple times). they should only be set to False if they cannot be computed.